### PR TITLE
Fix missing completion for unnamed symbols

### LINKED ISSUE #52155: Operator conversion completion does not show if followed by `var` statement
image

### PR DESCRIPTION
**Pull request by MaStr11**
Fix #52155

### CODE REVIEW

File: src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationServiceRunner.cs [hunk 1]
```diff
@@ -491,16 +491,16 @@ private RecommendedSymbols GetSymbolsOffOfConditionalReceiver(ExpressionSyntax o
             // applicable to it as well.
             var unnamedSymbols = _context.IsNameOfContext || excludeInstance
                 ? default
-                : GetUnnamedSymbols(originalExpression);
+                : GetUnnamedSymbols(containerSymbol, originalExpression);
```
Comment (MaStr11) [inline src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationServiceRunner.cs:494]: `containerSymbol` was found by speculation and needs to be passed. Here the speculation takes place:

https://github.com/dotnet/roslyn/blob/4ab21f1b9b6125c783531ab7a0d2479efd5fe306/src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationServiceRunner.cs#L280-L284

This is the explanation given there:
```CS
            // Check if we're in an interesting situation like this:
            //
            //     int i = 5;
            //     i.          // <-- here
            //     List<string> ml = new List<string>();
            //
            // The problem is that "i.List<string>" gets parsed as a type.  In this case we need 
            // to try binding again as if "i" is an expression and not a type.  In order to do 
            // that, we need to speculate as to what 'i' meant if it wasn't part of a local 
            // declaration's type.
            //
            // Another interesting case is something like:
            //
            //      stringList.
            //      await Test2();
            //
            // Here "stringList.await" is thought of as the return type of a local function.
```

File: src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationServiceRunner.cs [hunk 2]
```diff
@@ -491,19 +491,18 @@ private RecommendedSymbols GetSymbolsOffOfConditionalReceiver(ExpressionSyntax o
...
             return new RecommendedSymbols(namedSymbols, unnamedSymbols);
         }
 
-        private ImmutableArray<ISymbol> GetUnnamedSymbols(ExpressionSyntax originalExpression)
+        private ImmutableArray<ISymbol> GetUnnamedSymbols(ISymbol containerSymbol, ExpressionSyntax originalExpression)
```
Comment (CyrusNajmabadi) [inline src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationServiceRunner.cs:498]: i don't think this is the right approach.  Let me find the approach i think we should take here.

Comment (CyrusNajmabadi) [reply src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationServiceRunner.cs:498]: @MaStr11  opened https://github.com/dotnet/roslyn/pull/52193 for this.

**Comment by MaStr11** [comment]
@CyrusNajmabadi I see. Closed in favor of  #52193

**Pull request closed without merging**